### DISCUSSION TITLE
Add automatic fixups to French in Weblate

### DISCUSSION THREAD

**Discussion by PhilC813** [libre-tube/LibreTube#6565] [Ideas]
### Describe your suggested feature

French typography has specific standards which differ from English. Due to the limitations of some keyboards, the wrong characters are often.

Currently, the Weblate translation tool doesn't automatically correct these characters, which can lead to inconsistencies in translated strings. This could be solved permanently with Weblate's [custom automatic fixups](https://docs.weblate.org/en/latest/admin/checks.html#custom-automatic-fixups) feature.

### Apostrophes
**English:** _Apostrophe_ `'` ([U+0027](https://www.compart.com/en/unicode/U+0027))
**French equivalent:** _Right Single Quotation Mark_ `’` ([U+2019](https://www.compart.com/en/unicode/U+2019))
Suggested regex search & replace: `(?<=\p{L})\s*'\s*(?=\p{L})` > `’`

### Quotation Marks
**English:** _Quotation Mark_ `"` ([U+0022](https://www.compart.com/en/unicode/U+0022))
**French equivalent:** _Left-Pointing Double Angle Quotation Mark_ `«` ([U+00AB](https://www.compart.com/en/unicode/U+00AB)) on opening, _Right-Pointing Double Angle Quotation Mark_ `»` ([U+00BB](https://www.compart.com/en/unicode/U+00BB)) on closing, and with both, _No-Break Space_ ` ` ([U+00A0](https://www.compart.com/en/unicode/U+00A0)) separating from the quoted text
Suggested regex search & replace: `"\s*([^"]*?)\s*"` > `« $1 »`

### Other details

This is not an important issue (at all), but it could help uniformity in translations. Thank you!

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
Pull requests welcome. (I assume the fixup scripts need to be added to our GitHub repo and not via the Weblate web UI directly?)

**Comment by PhilC813** [comment]
The customization files must be added into the Python module containing the Weblate customization, and then you must add the fully-qualified path within the `settings.py` file.

I'm honestly not at ease with making changes with my current coding knowledge level. Weblate already has built-in automatic fixes, including for French specifically, so this feedback is probably more suitable for their project directly.

**Comment by PhilC813** [comment]
I've corrected all strings with the "Search and replace" tool of Weblate's Web UI for the time being
